packages/SwingSet/src/kernel/kernel.js

```diff
@@ -918,7 +918,7 @@ export default function buildKernel(
     // cleanup done, now we reset the worker to a clean state with no
     // transcript or snapshot and prime everything for the next incarnation.
 
-    await vatWarehouse.resetWorker(vatID);
+    const newIncarnation = await vatWarehouse.resetWorker(vatID);
```

Comment:
let's maybe rename `vatWarehouse.resetWorker` to `beginNewWorkerIncarnation`, since the side-effecty `incarnation`-bumping state changes aren't as obvious from the old name